draft-ietf-tls-ctls.md

```diff
@@ -139,7 +136,52 @@ where the most complexity and size reduction can be obtained. Most of the other
 TLS 1.3 are highly optimized and do not require compression to be used.
 
 The compression profile defining the use of algorithms, algorithm parameters, and
-extensions is specified via a JSON dictionary.
+extensions is represented by the `CTLSTemplate` structure:
+
+~~~~
+enum {
+  profile(0),
+  version(1),
+  cipher_suite(2),
+  dh_group(3),
+  signature_algorithm(4),
+  random(5),
+  mutual_auth(6),
+  handshake_framing(7),
+  client_hello_extensions(8),
+  server_hello_extensions(9),
+  encrypted_extensions(10),
+  cert_request_extensions(11),
+  known_certificates(12),
+  finished_size(13),
+  optional(65535)
+} CTLSTemplateElementType;
+
+struct {
+  CTLSTemplateElementType type;
+  opaque data<0..2^32-1>;
+} CTLSTemplateElement;
+
+struct {
+  CTLSTemplateElement elements<0..2^32-1>
+} CTLSTemplate;
+~~~~
+
+> TODO: Reorder enum.
+
+Elements in a `CTLSTemplate` MUST appear in strictly ascending order.
+The initial elements are defined in the subsections below.  Future elements can be
+added via an IANA registry ({{template-keys}}).  When generating a
+template, all elements are OPTIONAL
+to include.  When processing a template, all elements are mandatory
+to understand (but see discussion of `optional` in {{optional}}).
+
+For ease of configuration, an equivalent JSON dictionary format is also defined.
+It consists of a dictionary whose keys are the name of each element type (converted
+from snake_case to camelCase), and whose values are a type-specific representation
+of the element intended to maximize legibility.
```

Comment:
Is this change separable?

Reply:
I'm not sure what you mean.  I've put up #61 as an intermediate step that introduces the binary format without changing the transcript.

If you mean creating a version that doesn't define a JSON format at all: that is possible, but I didn't want to remove the JSON option without asking.  I have no strong preference on whether we define a JSON format.